todo/src/reducers/todos.js

```diff
@@ -0,0 +1,22 @@
+import { ADDTODO, REMOVETODO, TOGGLETODO } from '../actions/todoActions';
+
+
+export default (todos = [], action) => {
+    
+    switch (action.type) {
+        case ADDTODO:   
+            todos = [...todos, action.payload];
+            return todos;
+        case REMOVETODO:
+        return [...todos.slice(0, action.payload), ...todos.slice(action.payload+1)];
+        case TOGGLETODO:
```

Comment:
Since we're adding the `id` to the `todo` object, we can compare this way `todo.id === action.payload`
```    case TOGGLETODO:
        return todos.map(todo =>
            todo.id === +action.payload ? {...todo, toggle: true } : todo)```

Reply:
I had to add a + to `action.payload` since `getAttribute('id')` was giving a string.